backend/cmd/headlamp.go

```diff
@@ -150,7 +150,7 @@ func getOidcCallbackURL(r *http.Request, config *HeadlampConfig) string {
 		switch {
 		case fwdProto != "":
 			urlScheme = fwdProto
-		case r.Host == "localhost:"+config.port:
+		case strings.HasPrefix(r.Host, "localhost:") || r.TLS == nil:
```

Comment:
The commit title/body doesn't tell much about why this is needed. i.e. what was wrong, what does this fix?

Reply:
Updated